rct229/rulesets/ashrae9012019/ruleset_functions/get_building_segment_swh_bat.py

```diff
@@ -0,0 +1,123 @@
+]
+
+
+def get_building_segment_swh_bat(rmd: dict, building_segment_id: str) -> str:
+    """
+    This function determines the SWH BAT for the given building segment.
+
+    Parameters
+    ----------
+    rmd: dict
+        RMD at RuleSetModelDescription level
+    building_segment_id: str
+        building segment id
+
+    Returns
+    -------
+    building_segment_swh_bat: dict
+        one of the ServiceWaterHeatingSpaceOptions2019ASHRAE901 options
+
+    """
+
+    building_segment = find_exactly_one_with_field_value(
+        "$.buildings[*].building_segments[*]",
+        "id",
+        building_segment_id,
+        rmd,
+    )
+
+    if building_segment.get("service_water_heating_building_area_type"):
+        building_segment_swh_bat = building_segment[
+            "service_water_heating_building_area_type"
+        ]
+    else:
+        swh_use_dict = {}
+        service_water_heating_use_ids = (
+            get_swh_uses_associated_with_each_building_segment(rmd, building_segment_id)
+        )
+
+        for swh_use_id in service_water_heating_use_ids:
+            swh_use = find_exactly_one_with_field_value(
+                f"$.zones[*].spaces[*].service_water_heating_uses[*]",
+                "id",
+                swh_use_id,
+                building_segment,
+            )
+
+            if swh_use.get("use_units") == SERVICE_WATERH_EATING_USE_UNIT.OTHER:
+                return RCTOutcomeLabel.UNDETERMINED
+
+            swh_use_energy_by_space = get_energy_required_to_heat_swh_use(
+                swh_use["id"], rmd, building_segment["id"], False
+            )
+
+            if swh_use.get("area_type"):
+                area_type = swh_use["area_type"]
+                swh_use_dict.setdefault(area_type, ZERO.ENERGY)
+                swh_use_dict[area_type] += sum(swh_use_energy_by_space.values())
+            else:
+                for space_id in swh_use_energy_by_space:
+                    space = find_exactly_one_with_field_value(
+                        "$.zones[*].spaces[*]",
+                        "id",
+                        space_id,
+                        building_segment,
+                    )
+                    if space.get("service_water_heating_building_area_type"):
+                        service_water_heating_bat = space[
+                            "service_water_heating_building_area_type"
+                        ]
+                        swh_use_dict.setdefault(service_water_heating_bat, ZERO.ENERGY)
+                        swh_use_dict[
+                            service_water_heating_bat
+                        ] += swh_use_energy_by_space[space_id]
+                    else:
+                        swh_use_dict.setdefault(
+                            RCTOutcomeLabel.UNDETERMINED, ZERO.ENERGY
+                        )
+                        swh_use_dict[
+                            RCTOutcomeLabel.UNDETERMINED
+                        ] += swh_use_energy_by_space[space_id]
```

Comment:
Let's not to use the RCTOutcomLabel UNDETERMINED in the function. This will cause cross dependency, not good for maintenance.
You can just put `undetermined` string as the key.

Reply:
Addressed.